components/public-api-server/pkg/server/server.go

```diff
@@ -75,11 +75,13 @@ func Start(logger *logrus.Entry, version string, cfg *config.Configuration) erro
 		}
 	}
 
+	var stateJWT *oidc.StateJWT
```

Comment:
If there's no secret, the `stateJWT` can now be nil. We're passing it into 
```
oidc.NewService(cfg.SessionServiceAddress, dbConn, cipherSet, stateJWT)
```
Where it will panic when we try to access it on the calls. Should we hard fail if we don't have the secret? How do we handle this case?

Reply:
Was about to raise this question as well. 

First of,  adding config for staging and prod in [ops](https://github.com/gitpod-io/ops/pull/8269).

>  Should we hard fail if we don't have the secret? How do we handle this case?

Failing the whole component is really bad, but doesn't seem to be wrong as this as crucial bits from security perspective, i.e. falling back to a default is a "no go".

OTOH, a second option might be to check for `stateJWT == nil` in handlers and fail late. That comes at the cost of monitoring, as we'd need to cover that independently. 

How was that discussed with PATs and Stripe integration?

Reply:
@easyCZ, I'm opting for _not_ failing hard for now. We can change later on. Reason: for now this is all experimental and behind the scene, so let's keep risk at minimum.

Reply:
Done. Now it should not panic, but any attempt to call the endpoints while the signing key is not provided will be cancelled with an error.

Reply:
IMO, it's better to hard-fail in this case, I'm happy to do a follow-up change. With Stripe, we didn't hard fail because we had the use-case that we couldn't control the stripe configuration but needed the rest of the server to work. But here, we get a better rollout signal when we rollout and miss something.

Reply:
@easyCZ, let's do in follow-ups, please!

What else is required to release the breaks on this?

Reply:
PR is approved, feel free to land